### ISSUE TITLE
/setup breaks upgrading from 2.1.9 to 2.2.1

### ISSUE THREAD

**Issue by treii28** [magento/magento2#12118]
I'm in the process of setting up a new magento 2 site. Initially I set up the latest download from the magento website (2.2.0 at the time) but learned that the template we wanted to use was configured for 2.1.9 but claimed it could be upgraded 2.2.x after the template was installed. So I rebuilt the site as 2.1.9 from scratch and just initiated the upgrade (via the setup wizard url).

It seems to have completed properly but then the setup page dies while loading displaying {{$state.current.title}} on the page tab and only a partial header with a circled '1' and {{menuState.title}} written underneath pointing at {{menuState.url}}

The console log shows an Uncaught error in angular somewhere.

```
Uncaught Error: [$injector:modulerr] http://errors.angularjs.org/1.2.17-build.178+sha.2406084/$injector/modulerr?p0=magentoSetup&p1=Error%3A%20%5B%24injector%3Amodulerr%5D%20http%3A%2F%2Ferrors.angularjs.org%2F1.2.17-build.178%2Bsha.2406084%2F%24injector%2Fmodulerr%3Fp0%3Dcomponent-grid%26p1%3DError%253A%2520%255B%2524injector%253Anomod%255D%2520http%253A%252F%252Ferrors.angularjs.org%252F1.2.17-build.178%252Bsha.2406084%252F%2524injector%252Fnomod%253Fp0%253Dcomponent-grid%250A%2520%2520%2520%2520at%2520https%253A%252F%252Fwww.ourwebsite.com%252Fsetup%252Fpub%252Fangular%252Fangular.min.js%253A6%253A472%250A%2520%2520%2520%2520at%2520https%253A%252F%252Fwww.ourwebsite.com%252Fsetup%252Fpub%252Fangular%252Fangular.min.js%253A20%253A260%250A%2520%2520%2520%2520at%2520https%253A%252F%252Fwww.ourwebsite.com%252Fsetup%252Fpub%252Fangular%252Fangular.min.js%253A21%253A262%250A%2520%2520%2520%2520at%2520https%253A%252F%252Fwww.ourwebsite.com%252Fsetup%252Fpub%252Fangular%252Fangular.min.js%253A33%253A1%250A%2520%2520%2520%2520at%2520Array.forEach%2520(%253Canonymous%253E)%250A%2520%2520%2520%2520at%2520q%2520(https%253A%252F%252Fwww.ourwebsite.com%252Fsetup%252Fpub%252Fangular%252Fangular.min.js%253A7%253A298)%250A%2520%2520%2520%2520at%2520e%2520(https%253A%252F%252Fwww.ourwebsite.com%252Fsetup%252Fpub%252Fangular%252Fangular.min.js%253A32%253A445)%250A%2520%2520%2520%2520at%2520https%253A%252F%252Fwww.ourwebsite.com%252Fsetup%252Fpub%252Fangular%252Fangular.min.js%253A33%253A18%250A%2520%2520%2520%2520at%2520Array.forEach%2520(%253Canonymous%253E)%250A%2520%2520%2520%2520at%2520q%2520(https%253A%252F%252Fwww.ourwebsite.com%252Fsetup%252Fpub%252Fangular%252Fangular.min.js%253A7%253A298)%0A%20%20%20%20at%20https%3A%2F%2Fwww.ourwebsite.com%2Fsetup%2Fpub%2Fangular%2Fangular.min.js%3A6%3A472%0A%20%20%20%20at%20https%3A%2F%2Fwww.ourwebsite.com%2Fsetup%2Fpub%2Fangular%2Fangular.min.js%3A33%3A332%0A%20%20%20%20at%20Array.forEach%20(%3Canonymous%3E)%0A%20%20%20%20at%20q%20(https%3A%2F%2Fwww.ourwebsite.com%2Fsetup%2Fpub%2Fangular%2Fangular.min.js%3A7%3A298)%0A%20%20%20%20at%20e%20(https%3A%2F%2Fwww.ourwebsite.com%2Fsetup%2Fpub%2Fangular%2Fangular.min.js%3A32%3A445)%0A%20%20%20%20at%20https%3A%2F%2Fwww.ourwebsite.com%2Fsetup%2Fpub%2Fangular%2Fangular.min.js%3A33%3A18%0A%20%20%20%20at%20Array.forEach%20(%3Canonymous%3E)%0A%20%20%20%20at%20q%20(https%3A%2F%2Fwww.ourwebsite.com%2Fsetup%2Fpub%2Fangular%2Fangular.min.js%3A7%3A298)%0A%20%20%20%20at%20e%20(https%3A%2F%2Fwww.ourwebsite.com%2Fsetup%2Fpub%2Fangular%2Fangular.min.js%3A32%3A445)%0A%20%20%20%20at%20dc%20(https%3A%2F%2Fwww.ourwebsite.com%2Fsetup%2Fpub%2Fangular%2Fangular.min.js%3A36%3A34)
    at angular.js:161
    at angular.js:3808
    at Array.forEach (<anonymous>)
    at q (angular.js:215)
    at e (angular.js:3482)
    at dc (angular.js:3991)
    at c (angular.js:1198)
    at cc (angular.js:1244)
    at Wc (angular.js:1182)
    at angular.js:19141

```

the admin interface seems to load normally, but trying to go to the main website suddenly threw the server load up to 12, 10, 7 and took over 8 1/2 minutes to stop loading, meanwhile apache2 and php process still remained at a server load of about 9 running well after the browser seemed to stop loading content.

After waiting about 20 more minutes and not seeing the processes abate in any way, I restarted apache2. At that point, the main page, although altered somewhat from prior to the upgrade seems to load quick enough, but now I can't seem to get the admin dashboard to load any more and it too spawns a bunch of apache2 and php processes pushing server load up above 7.

I just need to get a working server up and going on the latest stable/secure version that will load the template we purchased and handle a plug-in we are about to purchase. This is a fresh install with barely anything at all modified. Why is it taking so long and why is the setup failing to load?

**Comment by treii28** [comment]
OK, experimenting around, I'm assuming the long-running apache processes were likely just static content, template, etc. rebuilding, although individual apache and php processes still occasionally spike high upon navigating around. Sometimes when I am not really doing anything as far as interacting with the browser at all.
I was able to get setup to fix itself by Ctrl-F5 (force reloading) the page in my browser. Still keeping an eye on performance issues to see if they eventually settle down.

**Comment by treii28** [comment]
Still having very slow responses, trying the various 'setup' left-bar options one by one. Most loaded and the processes settled down in under a minute, but the #system-config (last/bottom) option kept running for over 10 minutes with the spinning animated 'waiting/loading' indicator. I did a ctrl-F5 on that and it too eventually reloaded.
Going back through the items again, they aren't taking as long to load but still take from 20 seconds to 2 minutes each to load and push server load up around 1.5-2.5 (especially long/high on the system-config which is again still spinning busy after 4-5 minutes)

**Comment by treii28** [comment]
definitely something going on there you might want to look into. The themes I had installed at the time btw included the Novetty theme from zooextension (off of themeforest / envato) and the ves_need (venus themes need) off of the magento marketplace. Otherwise, there was very little done in the configuration menus beyond changing one or two default text items on the content pages.
Since it is a new install, I tried downloading a 2.2.1 package itself, backed up the existing folder and database (again) and re-installed from scratch and it isn't behaving anywhere close to as slow or fudgy as after trying that upgrade.
